Source/Controllers/SPAppController.m

```diff
@@ -705,6 +705,68 @@ - (void)openUserBundleAtPath:(NSString *)filePath
 		}
 	}
 
+	NSMutableArray *filesContainingLegacyString = [NSMutableArray array];
+
+	// enumerate dir
+	NSDirectoryEnumerator *enumerator = [fileManager
+										 enumeratorAtURL:[NSURL fileURLWithPath:filePath]
+										 includingPropertiesForKeys:@[NSURLIsRegularFileKey]
+										 options:NSDirectoryEnumerationSkipsHiddenFiles
+										 errorHandler:nil];
+
+	// check each file for legacy sequelpro string
+	for (NSURL *fileURL in enumerator) {
+		// Read the contents of the file into a string.
+		NSError *error = nil;
+		NSString *fileContentsString = [NSString stringWithContentsOfURL:fileURL
+																encoding:NSUTF8StringEncoding
+																   error:&error];
+
+		// Make sure that the file has been read, log an error if it hasn't.
+		if (!fileContentsString) {
+			SPLog(@"Error reading file");
+			continue;
+		}
+
+		// the string to search for
+		NSString *sequelpro = @"sequelpro";
+
+		// Search the file contents for the given string, put the results into an NSRange structure
+		NSRange result = [fileContentsString rangeOfString:sequelpro];
+
+		// -rangeOfString returns the location of the string NSRange.location or NSNotFound.
+		if (result.location == NSNotFound) {
+			SPLog(@"sequelpro NOT found in file: %@", fileURL.absoluteString);
+		}
+		else{
+			SPLog(@"sequelpro found in file: %@", fileURL.absoluteString);
+			SPLog(@"match: %@", [fileContentsString substringWithRange:result]);
+			SPLog(@"result: %lu, %lu", result.location, result.length);
+			[filesContainingLegacyString addObject:fileURL.absoluteString.lastPathComponent];
+		}
+	}
+
+	BOOL __block retCode = YES;
+
+	if(filesContainingLegacyString.count > 0){
+		NSString *filesString = [[filesContainingLegacyString valueForKey:@"description"] componentsJoinedByString:@"\n"];
+
+		[NSAlert createDefaultAlertWithTitle:[NSString stringWithFormat:NSLocalizedString(@"‘%@’ Bundle contains legacy components", @"Bundle contains legacy components"), filePath.lastPathComponent]
+									 message:[NSString stringWithFormat:NSLocalizedString(@"In these files:\n\n%@\n\nDo you still want to install the bundle?", @"Do you want to install the bundle?"), filesString]
+						  primaryButtonTitle:NSLocalizedString(@"Install", @"Install")
+						primaryButtonHandler:^{
+			SPLog(@"Continue, install");
```

Comment:
I think the only change I would make here would be any legacy bundle the option should be "attempt to upgrade and install" and we should replace all occurrences of sequelpro with sequelace. I can't think of a use case where the user wouldn't want the bundle to just be updated to work with ace if possible? I think the warning is still good for letting them know it may be unsupported

Reply:
I also don't think it would be bad if Ace worked with exclusively .sabundle files and could open .spbundle files but would convert them to .sabundle when opened/added

Reply:
> I also don't think it would be bad if Ace worked with exclusively .sabundle files and could open .spbundle files but would convert them to .sabundle when opened/added

I've got this part working.

Reply:
Still need to handle:

- Saving an `.spBundle` change to `.saBundle`
- Migrate any user `spBundles` already in the folder
- Attempt to replace legacy strings e.g. `sequelpro`